cognite/client/data_classes/_base.py

```diff
@@ -580,7 +580,7 @@ def __init__(
         geometries: Collection[Geometry] | None = None,
     ) -> None:
         if type not in self._VALID_TYPES:
-            raise ValueError(f"type must be one of {self._VALID_TYPES}")
+            raise TypeError(f"type must be one of {self._VALID_TYPES}")
```

Comment:
This should stay as type is annotated using `Literal[...]` only 😄 
```suggestion
            raise ValueError(f"type must be one of {self._VALID_TYPES}")
```

Reply:
Aha, first now I think I get this `Literal` thing - thanks!